### PR TITLE
Add .NET 8 target (Avalonia 11.1)

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?

There is a number of features that we need from .NET 8 SDK and targets.
Primarily:
1. Keep mobile support up to date. Xamarin team actively drops support for .NET TFM versions, and right now it's not possible to build mobile app with .NET 6 TFM. .NET 7 still works, but this PR is going one step ahead.
2. `UnsafeAccessor` which is going to be helpful with SkiaSharp 3.0 compatibility
3. Up-to-date SDK analyzers.
4. Other potential optimizations available only on .NET 8 SDK. Though we still can't use .NET 8 only features in our Public API (like, DateOnly primitives), but it can go to Avalonia.Labs.

And also, we are likely to drop .NET 6 in Avalonia 12.0 (while still likely to support .NET Standard 2.0). Targeting .NET 8 is a preparation before that.

### CODE REVIEW

**Comment by workgroupengineering** [comment]
Instead of directly using TargetFramework `net8.0` for each project, add to `Directory.Build.props` a property like

```xml
<PropertyGroup>
  <AvaloniaDefaultTargetFramework Condition="'$(AvaloniaDefaultTargetFramework)' == ''">net8.0</AvaloniaDefaultTargetFramework>
</PropertyGroup>
``` 

eg.  using in `packages/Avalonia/Avalonia.csproj`

```xml
  <PropertyGroup>
      <TargetFrameworks>$(AvaloniaDefaultTargetFramework);netstandard2.0</TargetFrameworks>
      <PackageId>Avalonia</PackageId>
  </PropertyGroup>
``` 

For example, when you switch to `net9.0` targte you will only need to change it in one point.

**Comment by maxkatz6** [comment]
Yeah, we discussed that internally.
I had an idea for DotNetCurrent=net8.0, DotNetPrevious=net6.0, DotNetFramework=net461, DotNetStandard=netstandard2.0.

But we couldn't find if it really would make anything better. It does make it easier to update version, but at the same time harder to quickly check csproj version for users no used to the repo. And in general, it's not like we change these versions often. Next time probably will happen around .NET 10 haha.

**Comment by kekekeks** [comment]
Can we have `<LegacyTargetFrameworks Condition="'$(AvaloniaSkipLegacyTargetFrameworks)' != 'True'>net6.0</LegacyTargetFrameworks>` so building for net6.0 can be disabled during development?

**Comment by maxkatz6** [comment]
@kekekeks it would complicate msbuild logic quite a bit. Will think about it. It would also add DotNetCurrent/DotNetPrevious props anyway, which is fine.

**Comment by kekekeks** [comment]
> it would complicate msbuild logic quite a bit

Why?
It would be something like this in common props:
```xml
<DotNetCurrentFramework>net9.0</DotNetCurrentFramework>
<DotNetCompatibilityFrameworks>net8.0;net6.0<DotNetCompatibilityFrameworks>
<CommonTargetFrameworks>$(DotNetCurrentFramework);netstandard2.0</CommonTargetFrameworks>
<CommonTargetFrameworks Condition="'$(AvaloniaSkipLegacyTargetFrameworks)' != 'True'>$(CommonTargetFrameworks);$(DotNetCompatibilityFrameworks)</CommonTargetFrameworks>
```

and  `<TargetFrameworks>$(CommonTargetFrameworks)</TargetFrameworks>` in csproj files.

**Comment by workgroupengineering** [comment]
If you decide to add properties, it may be best to add the Avalonia prefix to the properties to try to avoid conflicts.

**Comment by maxkatz6** [comment]
> It would be something like this in common props:

So, how would we add "-ios", "-android", "-browser" suffix?

**Comment by kekekeks** [comment]
OK, it's out of the scope of this PR anyway.

File: src/Browser/Avalonia.Browser.Blazor/Avalonia.Browser.Blazor.csproj [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
   <PropertyGroup>
     <TargetFramework>net7.0</TargetFramework>
```
Comment (MrJul) [inline src/Browser/Avalonia.Browser.Blazor/Avalonia.Browser.Blazor.csproj:4]: Can be removed, the property is redefined to `net8.0-browser` at the next line.

File: src/Browser/Avalonia.Browser/Avalonia.Browser.csproj [hunk 2]
```diff
@@ -1,6 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>net7.0</TargetFramework>
```
Comment (MrJul) [inline src/Browser/Avalonia.Browser/Avalonia.Browser.csproj:3]: Can be removed, the property is redefined to `net8.0-browser` at the next line.

File: src/iOS/Avalonia.iOS/Avalonia.iOS.csproj [hunk 3]
```diff
@@ -1,11 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net7.0-ios16.0;net7.0-tvos</TargetFrameworks>
+    <TargetFrameworks>net8.0-ios17.0;net8.0-tvos17</TargetFrameworks>
```
Comment (kekekeks) [inline src/iOS/Avalonia.iOS/Avalonia.iOS.csproj:3]: Why are we bumping the minimum supported version of iOS?

**Review by grokys — APPROVED** [review]

**Pull request merged**